infra/modules/database/main.tf

```diff
@@ -1,23 +1,68 @@
 data "aws_caller_identity" "current" {}
 data "aws_region" "current" {}
+data "aws_vpc" "default" {
+  default = true
+}
+
+locals {
+  admin_user                 = "app_usr"
+  admin_user_secret_name     = "/metadata/db/${var.database_name}-admin-user"
+  admin_password_secret_name = "/metadata/db/${var.database_name}-admin-password"
+  admin_db_url_secret_name   = "/metadata/db/${var.database_name}-admin-db-url"
+  database_name_formatted    = replace("${var.database_name}", "-", "_")
+}
+
+####################
+## Security Group ##
+####################
+resource "aws_security_group" "database" {
+  # Specify name_prefix instead of name because when a change requires creating a new
+  # security group, sometimes the change requires the new security group to be created
+  # before the old one is destroyed. In this situation, the new one needs a unique name
+  name_prefix = "${var.database_name}-database"
+  description = "Allow inbound TCP access to database port"
+  vpc_id      = data.aws_vpc.default.id
+
+  lifecycle {
+    create_before_destroy = true
+  }
 
+  ingress {
+    description = "Allow HTTP traffic to database port"
+    protocol    = "tcp"
+    from_port   = var.database_port
+    to_port     = var.database_port
+    cidr_blocks = [data.aws_vpc.default.cidr_block]
+  }
 
-###########################
+  egress {
+    description = "Allow all outgoing traffic from database"
+    protocol    = "-1"
+    from_port   = 0
+    to_port     = 0
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+}
```

Comment:
This is unnecessary.
imo You shouldn't let egress traffic leave the VPC for the database